code/modules/mob/living/simple_animal/abnormality/he/red_queen.dm

```diff
@@ -17,20 +18,93 @@
 	work_damage_amount = 20			//Unlikely to hurt you but if she ever does she'll fuck you
 	work_damage_type = RED_DAMAGE
 
+	ranged = TRUE
+	minimum_distance = 2
+	retreat_distance = 2
+	move_to_delay = 5
+	stat_attack = HARD_CRIT
+
 	ego_list = list(
 		/datum/ego_datum/weapon/fury,
 		/datum/ego_datum/armor/fury
 		)
 	gift_type = /datum/ego_gifts/fury
 
 	var/liked
+	var/list/diamonds = list()
+	var/diamond_cooldown = 2 SECONDS
+	COOLDOWN_DECLARE(diamond)
 
 /mob/living/simple_animal/hostile/abnormality/red_queen/Initialize(mapload)
 	. = ..()
 	//What does she like?
 	//Pick it once so people can find out
 	liked = pick(ABNORMALITY_WORK_INSTINCT, ABNORMALITY_WORK_INSIGHT, ABNORMALITY_WORK_ATTACHMENT, ABNORMALITY_WORK_REPRESSION)
 
+/mob/living/simple_animal/hostile/abnormality/red_queen/Moved()
+	. = ..()
+	var/valid_turf = shuffle(view(2, src))
+	for(var/obj/projectile/queen_diamond/QD in diamonds)
+		for(var/turf/open/T in valid_turf)
+			if(T == get_turf(src))
+				continue
+			var/filled = FALSE
+			for(var/obj/projectile/P in T.contents)
```

Comment:
Couldn't you just get turfs of the diamonds since you are already iterating through them?